src/zenml/stack/stack.py

```diff
@@ -751,6 +751,42 @@ def prepare_step_run(self, info: "StepRunInfo") -> None:
         ).values():
             component.prepare_step_run(info=info)
 
+    def publish_pipeline_run_metadata(self, run_id: UUID) -> None:
```

Comment:
Any reason these methods can't be inside the step launcher/runner to keep them in the same place as all the other client calls related to pipeline/step runs? They seem a little out of place here in my opinion

Reply:
I agree. Thinking about it again, I would say the cleanest design would be to have the stack collect the entire metadata as a dict and then have the client call to publish it somewhere in the launcher/runner. Will look into this again 👍